source/postprocess/boundary_strain_rate_residual_statistics.cc

```diff
@@ -0,0 +1,333 @@
+                             "The name of a directory that contains the ascii data. "
+                             "This path may either be absolute (if starting with a "
+                             "`/') or relative to the current directory. The path may also "
+                             "include the special text `$ASPECT_SOURCE_DIR' which will be "
+                             "interpreted as the path in which the ASPECT source files were "
+                             "located when ASPECT was compiled. This interpretation allows, "
+                             "for example, to reference files located in the `data/' subdirectory "
+                             "of ASPECT.");
+          prm.declare_entry ("Data file name", "box_3d_boundary_strain_rate.txt",
+                             Patterns::Anything (),
+                             "The file name of the input surface strain rate an ascii data. "
+                             "The file has (dim-1)*(dim-1) columns in addition to the coordinate columns. "
+                             "The strain rate data columns correspond to the different strain rate components "
+                             "at the surface, i.e. 2 components for 2d models and 4 components for 3d models. ");
+          prm.declare_entry ("Scale factor", "1.",
+                             Patterns::Double (),
+                             "Scalar factor, which is applied to the model data. "
+                             "You might want to use this to scale the input to a "
+                             "reference model.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    BoundaryStrainRateResidualStatistics<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Boundary strain rate residual statistics");
+        {
+          // Get the path to the data files. If it contains a reference
+          // to $ASPECT_SOURCE_DIR, replace it by what CMake has given us
+          // as a #define
+          data_directory = Utilities::expand_ASPECT_SOURCE_DIR(prm.get ("Data directory"));
+          data_file_name    = prm.get ("Data file name");
+          scale_factor      = prm.get_double ("Scale factor");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    ASPECT_REGISTER_POSTPROCESSOR(BoundaryStrainRateResidualStatistics,
+                                  "boundary strain rate residual statistics",
+                                  "A postprocessor that computes some statistics about "
+                                  "the surface strain rate residual along the top boundary. The residual "
+                                  "is the difference between the second invariant of model solution gradients and "
```

Comment:
How the strain rate is calculated is not as important here, it is more important to stay consistent with the rest of the documentation.
```suggestion
                                  "is the difference between the second invariant of the model strain rate and "
```